src/Database/DBA.php

```diff
@@ -251,22 +255,20 @@ private static function logIndex($query) {
 	}
 
 	public static function escape($str) {
-		if (self::$connected) {
-			switch (self::$driver) {
-				case 'pdo':
-					return substr(@self::$connection->quote($str, PDO::PARAM_STR), 1, -1);
-				case 'mysqli':
-					return @self::$connection->real_escape_string($str);
-			}
-		} else {
-			return str_replace("'", "\\'", $str);
+		switch (self::$driver) {
+			case 'pdo':
+				return substr(@self::$connection->quote($str, PDO::PARAM_STR), 1, -1);
+
+			case 'mysqli':
+				return @self::$connection->real_escape_string($str);
 		}
 	}
 
 	public static function connected() {
 		$connected = false;
 
 		if (is_null(self::$connection)) {
+			/// @TODO Shouldn't we better prevent this incorrect invocation rather than failing silently?
```

Comment:
In no case we should create an error here. We are calling this function for example from the connect function. With raising an error here we couldn't create a database connection anymore at all.